pkg/k8sutil/create.go

```diff
@@ -227,3 +230,93 @@ func createNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) erro
 func ListNamespaces(nsclient corev1typed.NamespaceInterface) (*v1.NamespaceList, error) {
 	return nsclient.List(context.TODO(), metav1.ListOptions{})
 }
+
+// ObjectMetadata uniquely identifies any object in the list of YAML manifests.
+type ObjectMetadata struct {
+	Version string
+	Kind    string
+	Name    string
+}
+
+// YAMLToUnstructured accepts a Kubernetes manifest in YAML format and returns an object of type
+// `unstructured.Unstructured`. This object has many methods that can be used by the consumer to
+// extract metadata from the Kubernetes manifest.
+func YAMLToUnstructured(yamlObj []byte) (*unstructured.Unstructured, error) {
+	u := &unstructured.Unstructured{}
+
+	// Decode YAML into `unstructured.Unstructured`.
+	dec := yamlserializer.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	if _, _, err := dec.Decode(yamlObj, nil, u); err != nil {
```

Comment:
Suggestion: We could make `nil` as `var gvk *schema.GroupVersionKind`, so it's obvious which value we pass nil without an IDE.